pennylane/tape/qscript.py

```diff
@@ -1410,3 +1411,76 @@ def wrapper(*args, **kwargs):
         return qscript
 
     return wrapper
+
+
+def bind_new_parameters_tape(
+    tape: QuantumScript, params: Sequence[TensorLike], indices: Sequence[int]
+):
+    """Create a new tape with updated parameters.
+
+    This function takes a :class:`~.tape.QuantumScript` as input, and returns
+    a new ``QuantumScript`` containing the new parameters at the provided indices,
+    with the parameters at all other indices remaining the same.
+
+    Args:
+        tape (.tape.QuantumScript): Tape to update
+        params (Sequence[TensorLike]): New parameters to create the tape with. This
+            must have the same length as ``indices``.
+        indices (Sequence[int]): The parameter indices to update with the given parameters.
+            The index of a parameter is defined as its index in ``tape.get_parameters()``.
+
+    Returns:
+        .tape.QuantumScript: New tape with updated parameters
+    """
+    # pylint: disable=no-member
+
+    if len(params) != len(indices):
+        raise ValueError("Number of provided parameters does not match number of indices")
+
+    new_ops = []
+    idx = 0
+    p_idx = 0
+
+    for op in tape.circuit:
+        if isinstance(op, Operator):
+            data = op.data
+        elif op.obs is not None:
+            data = op.obs.data
+        else:
+            data = ()
+
+        # determine if any parameters of the operator need to be rebinded
+        if any(i + idx in indices for i in range(len(data))):
+            new_params = []
+            for i, d in enumerate(data):
+                if i + idx in indices:
+                    new_params.append(params[p_idx])
+                    p_idx += 1
+                else:
+                    new_params.append(d)
+
+            if isinstance(op, Operator):
+                new_op = qml.ops.functions.bind_new_parameters(op, new_params)
+            else:
+                new_obs = qml.ops.functions.bind_new_parameters(op.obs, new_params)
+                new_op = op.__class__(obs=new_obs)
+
+            new_ops.append(new_op)
+        else:
+            # no need to change the operator
+            new_ops.append(op)
+
+        idx += len(data)
+
+    new_prep = new_ops[: len(tape._prep)]
+    new_operations = new_ops[len(tape._prep) : len(tape.operations)]
+    new_measurements = new_ops[len(tape.operations) :]
+
+    new_tape = qml.tape.QuantumScript(new_operations, new_measurements, new_prep, shots=tape.shots)
+    new_tape.trainable_params = tape.trainable_params
+    new_tape._qfunc_output = tape._qfunc_output
+
+    return new_tape
+
+
+QuantumScript.bind_new_parameters = bind_new_parameters_tape
```

Comment:
So this functionality should be either a function or a method, not both. 

Pros of function:
* Easier to hide away

Pros of method:
* we don't have to import the function, as it already exists on the object.

We should also modify the same in such a way that it is clearly distinguished from `bind_new_parameters`, in that it can accept a specific subset of indices. But `bind_new_parameters_at_indices` is too long. 🤔 

Also, `bind_new_parameters` itself will be able to be redone following the completion of pytrees. `bind_new_parameters_at_indices` will not.

OK, I'm way overthinking this.

1) Let's just have it as a method, not a standalone function.
2) Let's just call it `bind_new_parameters`, even though it differs from `functions.bind_new_parameters`

Reply:
Updated 👍